### PR TITLE
Go home, grandpa

### LINKED ISSUE #1921: Signup Data Form: Old people messaging
- [x] Create a `dosomething_user_is_old_person` function which returns TRUE if user is over 25
- [x] Create a new property for a signup data form, "Prevent old people from submitting", which will not be checked by default.
- [x] Modify the auto-prompt user (refs #1913) functionality to check for this setting, and if set, auto-prompt based on whether not `dosomething_user_is_old_person`
- [x] Create a new property for "Old people copy", which is only visible in the config form when "Prevent old people from submitting" is checked
- [x] Modify the user-facing signup data form to display the "Old people copy" if an old person is viewing the signup data form (which would be done by clicking the modal link in the action page)

### PR DESCRIPTION
**Pull request by aaronschachter**
@angaither Can you please review?

Fixes #1921.

Adds two new properties to the Signup Data Form configuration:
- Prevent old people from submitting
- Old people copy

Displays old people copy if an old person clicks on the modal form.  Also makes sure not to auto-prompt an old person if the form is required.

### CODE REVIEW

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module [hunk 1]
```diff
@@ -330,6 +331,19 @@ function dosomething_campaign_add_signup_data_form_vars(&$node) {
     );
     return;
   }
+
+  // If form is set to prevent Old People from submitting, and you're Old:
+  if ($config['prevent_old_people_submit'] && dosomething_user_is_old_person()) {
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module:336]: do we want to ignore if `dosomething_user_is_staff` ?

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc [hunk 2]
```diff
@@ -80,6 +80,24 @@ function _dosomething_signup_node_signup_data_form(&$form, &$form_state) {
       ),
     ),
   );
+  $form[$fieldset]['config'][$prefix . 'prevent_old_people_submit'] = array(
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc:83]: `$prefix . '_prevent_old_people_submit'`

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc [hunk 3]
```diff
@@ -80,6 +80,24 @@ function _dosomething_signup_node_signup_data_form(&$form, &$form_state) {
...
+    '#type' => 'checkbox',
+    '#title' => t('Prevent Old People from submitting'),
+    '#default_value' => $values['prevent_old_people_submit'],
+    '#description' => t('If checked, Old People will be unable to submit the form.'),
+  );
+  $form[$fieldset]['config'][$prefix . 'old_people_copy'] = array(
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc:89]: same

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module (hunk 1 above)
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module:336]: I think that would make things a little more confusing for staff.  if they get an old person message, they'll know that things are working properly, and also have the system knowledge to know to change their birthday to test.

File: lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc (hunk 2 above)
Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc:83]: `$prefix` does include a `_`: See https://github.com/aaronschachter/dosomething/blob/old_person/lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc#L35

Comment (angaither) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc:83]: That's pretty confusing... do we normally do it like that?

Comment (aaronschachter) [inline lib/modules/dosomething/dosomething_signup/dosomething_signup.signup_data_form.inc:83]: It's always been like that in this function at least... and similar to `dosomething_helpers_get_custom_var_prefix` as well, where the '_` is included in the prefix.

**Comment by angaither** [comment]
:+1:

**Pull request merged**